### ISSUE TITLE
Profile badges are distorted

### ISSUE THREAD

**Issue by haukist** [signalapp/Signal-Android#11809]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read -  https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
 The new profile badges are distorted.

### Screenshots

![Screenshot_20211204-133239.png](https://user-images.githubusercontent.com/14955895/144697512-4b19a87b-bb2d-4e11-b9a7-e5d8a665cb3e.png)

### Device info

**Device:** Google Pixel 6
**Android version:** 12 
**Signal version:** 5.27.13

**Comment by alex-signal** [comment]
Hi! Could you please try clearing cache and seeing if that resolves the issue?

(Do NOT clear data, that'll wipe your Signal install)

**Comment by MirzaD** [comment]
Clearing cache didn't solve the problem for me. Cleared cache, force closed the app and opened again, still distorted

**Comment by alex-signal** [comment]
Ok thanks! I'll have to likely add a bit of logging to see which density of the badges your app is downloading =/

**Comment by haukist** [comment]
Yeah cleaning cache didn't make a difference. I see the same problem with other Android phones too.

**Comment by haukist** [comment]
This affects supporter satisfaction. Please fix it.

**Comment by cody-signal** [comment]
Hi there, can you submit a debuglog please? It'll have information about the display settings that can help us figure out what's going on here. Thanks!

**Comment by haukist** [comment]
Here is my debug log. Thanks.
https://debuglogs.org/android/5.34.5/30c8c757ab8016553a975b1d3cdda1b2cbc85a3d56b17731a5642a0ed09014ae

**Comment by nonsenseninja** [comment]
Experiencing the exact same issue on my Pixel 3a. Please take a look at my debug log if it helps.
https://debuglogs.org/android/5.37.4/0c33b44b413c2db2ea1a7d8e20dd713e0269fb35f5d543620b8b100535c4f3c7

**Comment by alex-signal** [comment]
Hi! Have you modified your display size at all in system settings? (Just trying to think through a couple different scenarios)

**Comment by nonsenseninja** [comment]
Hi, I'm not the other guy but I also still encounter this issue on my Pixel 3a and my display size is set to default.

**Comment by alex-signal** [comment]
Thanks for the update. I believe we're going to be shipping an update with some additional logging so we can see exactly which density bucket it's downloaded.

Q for the crowd, is it *all* instances of badges or only in certain places that look aliased / distorted? It could just be bad scaling (which would have a separate fix compared to if we were downloading an incorrect density)

**Comment by MirzaD** [comment]
There seems to be a distroton on both places where I can see my badge. Adding zoomed in screenshots to showcase.
![Screenshot_20220719-164407](https://user-images.githubusercontent.com/5558026/179779425-a2d363d8-753e-4023-8b96-40dcd28302ce.jpg)
![Screenshot_20220719-164353](https://user-images.githubusercontent.com/5558026/179779444-e2ad30a3-3ac4-4272-8808-864a98b9081e.jpg)

**Comment by haukist** [comment]
It seems all instances of badges look distorted. Only the animated rocket badge on the One-time Donation screen looks fine.

**Comment by alex-signal** [comment]
Okie dokie. So it turns out that while Pixel 3 reports XXHDPI (3x) it's actual display density is 2.75x. Thus you end up with a bunch of compression aliasing.

I just added a change which'll ship to the next beta which will perform a filtered bitmap scale when there's a mismatch between the sprite size and the target output size. Initial checks (using a Pixel 3 emulator) look great.

**Comment by nonsenseninja** [comment]
It's fixed on my end! Thanks!
